doc/index.md

```diff
@@ -1032,68 +1050,153 @@ On success, returns the response [*headers*](#http.headers) and a [*stream*](#st
+  -         `false`: Allows non-tls connections only
+  - `.version`: The http version to allow to connect (default: any)
+
+
+#### Example {#http.server.listen-example}
 
-### `listen(options)` <!-- --> {#http.server.connect}
+```lua
+
+local app_server = http_server.listen {
+		host = 192.168.0.1;
+		port = 80;
+		onstream = function(server, stream)  
+			local request_headers = assert(stream:get_headers())
+			local request_method = request_headers:get ":method"
+
+			print(string.format('[%s] "%s %s HTTP/%g"  "%s" "%s" ',
+				os.date("%d/%b/%Y:%H:%M:%S %z"),
+				request_headers:get(":method") or "",
+				request_headers:get(":path") or "",
+				stream.connection.version,
+				request_headers:get("referer") or "-",
+				request_headers:get("user-agent") or "-"
+				))
+		end;
+}
+
+```
 
 
 ### `server:onerror(new_handler)` <!-- --> {#http.server:onerror}
 
+If called with parameters, the function replaces the current error handler function with `new_handler`  and returns a reference to the old function. Calling the function with no parameters returns the current error handler. The default error handler is to do nothing, and can be set during instantiation through the `options` table passed to the [*server.listen(options)*](#server.listen) function.
+
 
 ### `server:listen(timeout)` <!-- --> {#http.server:listen}
 
+Initializes the server socket and if required, resolves DNS and instantiates TLS. `server:listen` is required if [*localname*](#http.server:localname) is called before [*step*](#http.server:step) or [*loop*](#http.server:loop). On error, returns nil, an error message and an error number.
+
 
 ### `server:localname()` <!-- --> {#http.server:localname}
 
+Returns the connection information for the local socket. Returns address family, IP address and port for an external socket. For Unix domain sockets, the function returns AF_UNIX and the path. If the connection object is not connected, returns AF_UNSPEC (0). On error, returns nil an error message and an error number.
+
 
 ### `server:pause()` <!-- --> {#http.server:pause}
 
-Cause the server loop to stop processing new clients until [`:resume`](#http.server:resume) is called.
+Cause the server loop to stop processing new clients until [`:resume`](#http.server:resume) is called. Existing client connections will run until closed. 
 
 
 ### `server:resume()` <!-- --> {#http.server:resume}
 
+Resumes a [*paused*](#http.server:pause) `server` and processes new client requests. 
+
 
 ### `server:close()` <!-- --> {#http.server:close}
 
+Shutdown the server and close the socket. A closed server cannot be reused.
+
 
 ### `server:pollfd()` <!-- --> {#http.server:pollfd}
 
+Returns a file descriptor (as an integer) or nil. The file descriptor can be passed to a system API like select or kqueue to wait on anything this server object wants to do. `pollfd` is used for integrating with other main loops, and should be used in combination with :events() and :timeout(). 
```

Comment:
select and kqueue should be in backticks. `:events()` and `:timeout()` should also be in backticks, and have links to their respective sections